lib/sqlalchemy_ingres/base.py

```diff
@@ -323,12 +330,15 @@ def fire_sequence(self, seq, type_):
 
     def pre_exec(self):
         if self.isinsert:
-            if TYPE_CHECKING:
-                assert is_sql_compiler(self.compiled)
-                assert isinstance(self.compiled.compile_state, DMLState)
-                assert isinstance(
-                    self.compiled.compile_state.dml_table, TableClause
-                )
+            if (sqlalchemy_version_tuple >= (2,0)):
+                from typing import TYPE_CHECKING
```

Comment:
Conditional imports are totally fine, pep8 says to do this at the head of the file. So all imports occur at the top for ease of reading/maintenance.

Recommend these be moved.

Navel gazing comments follow.

There are a few approaches to conditional imports:

1. Import error reactive, in a try/except block with NEW import performed first (rather than old version)
      * possibly with a "flag" to clarify what has been import. For example in failing case can set module name to None (possible run time errors if check later missing)
2. version specific check like you have - recommend against UNLESS there is no other option
      * library/package version
      * Python version
3. capability/platform check, for example; is this Windows/Linux, is this CPython, is this PyPy, is this Jython, ....

You may... have to conditionals later "at execution time" that depend on which import succeeded.

In this example you already have runtime checks. Proposal:

    try:
        from typing import TYPE_CHECKING
    expect ImportError:
        TYPE_CHECKING = None

Reply:
Improved handling of conditional import and use of version specific SQLAlchemy function `is_sql_compiler`.